test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -1806,8 +1805,8 @@ protected TestCluster buildTestCluster(Scope scope, long seed) throws IOExceptio
             ArrayList<Class<? extends Plugin>> mocks = new ArrayList<>(mockPlugins);
             // add both mock plugins - local and tcp if they are not there
             // we do this in case somebody overrides getMockPlugins and misses to call super
-            if (mockPlugins.contains(MockTcpTransportPlugin.class) == false) {
-                mocks.add(MockTcpTransportPlugin.class);
+            if (mockPlugins.contains(randomTestTransportPlugin()) == false) {
+                mocks.add(randomTestTransportPlugin());
```

Comment:
Doesn't this need to match up with the transport key specified in getNodeConfigSource below?